app/src/main/java/com/kamron/pogoiv/OcrHelper.java

```diff
@@ -336,12 +336,27 @@ private String getCandyNameFromImg(Bitmap pokemonImage) {
 
 
     /**
-     * Gets the candy amount from a pokenon image.
+     * Gets the candy amount from a pokemon image.
      *
      * @param pokemonImage the image of the whole screen
      * @return candyAmount the candy amount, or blank Optional object if nothing was found
      */
     private Optional<Integer> getCandyAmountFromImg(Bitmap pokemonImage) {
+
+        //disable getting CandyAmount if we dont have pokespam running
+        GoIVSettings Settings;
+        try {
+            //Anything better then getInstance(null)? should we let it crash?
+            Settings = GoIVSettings.getInstance(null);
+        } catch (Exception ex) {
+            //probably a crash due to getInstance
+            return Optional.absent();
+        }
+
+        if (Settings == null || !Settings.isPokeSpamEnabled()) {
```

Comment:
Remember local variables start lowercase, uppercase is for classnames, so `Settings.isPokeSpamEnabled` alone looks like a static method call...

Re `getInstance(null)`, I'll see what we can do, but a crash should not be possible.